code-challenges.js

```diff
@@ -45,10 +106,79 @@ var stringWithVowels2 = "ILoveJavaScript"
 
 
 
-
-// --------------------6) Expand the logic from #4 to inform your user if the variable passed into the vowel removal function is not a string.
+//create a function
+const noVowels = (string) => {
+    //created a new variable no store our string with removed vowels.
+    
+    //string.replace we enter what we want to be replaced in between two /'s. It can be a word or single characters separated by |, in this case we used | to separate vowels. 
+    
+    //The argument following the / / can either be g (global), or gi (global case-insensitive). Used gi to cover uppercase vowels as well.
+    
+    //Then it is followed by what we want it replaced with. In this case nothing so we put ""
+    let nVwls = string.replace(/a|e|i|o|u/gi, "")
+    //return the string with no vowels
+    return nVwls
+}
+//call the function and enter argument here.
+// console.log(noVowels(stringWithVowels1));
+// console.log(noVowels(stringWithVowels2));
+
+//IGNORE BELOW IGNORE BELOW IGNORE BELOW IGNORE BELOW
+//following along with the class on this block of code, alternate way of solving problem
+
+const noVowelsPlease = (string) => {
+    let vowelArray = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
+    let charArr = string.split("")
+    let noVowelArray = charArr.filter((value, index) => {
+        let isValue = true
+        return isValue !== vowelArray.includes(charArr[index])
+    })
+let noVowelString = noVowelArray.join("")
+return noVowelString
+}
+// console.log(noVowelsPlease(stringWithVowels1));
+// console.log(noVowelsPlease(stringWithVowels2));
+
+// const zeroVowels = (str) => {
+//     let arrayVowel = str.split("")
+//     arrayVowel.filter(value =>{
+//         value != "a" &&
+//     }).join("")
+// }
+
+//IGNORE ABOVE IGNORE ABOVE IGNORE ABOVE IGNORE ABOVE
+
+
+// --------------------6) Expand the logic from #5 to inform your user if the variable passed into the vowel removal function is not a string.
 // Copy and paste or rewrite the function. Use the test variables provided below.
 
 var stringWithVowels3 = true
 var stringWithVowels4 = 42
 var stringWithVowels5 = "CodingIsGreat"
+
+const noVowelsExpanded = (string) => {
+    //created a new conditional that checks the data type, if it is a "string" it moves on to remove vowels.
+    if(typeof string === "string") {
```

Comment:
HECK YES. Good work! Great refactoring and adding in the conditional to reach your solution!